.gitlab/pipeline.yml

```diff
@@ -10,56 +10,140 @@ include:
+    DOCKERFILE: .gitlab/docker/Dockerfile.spack_base
+    DOCKER_BUILD_ARGS: '["BASE_IMAGE","SPACK_VERSION"]'
+    CSCS_REBUILD_POLICY: always
+
+###################################################################################################
+
+gcc11_spack_image:
+  stage: spack_configs
+  needs: [ base_spack_image ]
+  extends:
+    - .container-builder
+    - .variables_gcc11_config
+  before_script:
+    - CONFIG_TAG=`echo $CXXSTD-$BOOST-$HWLOC | sha256sum - | head -c 16`
+    - compiler=${COMPILER/@/-}
+    - export PERSIST_IMAGE_NAME=$CSCS_REGISTRY_PATH/pika-$compiler:$CONFIG_TAG
+    - echo -e "compiler=$compiler\nBASE_IMAGE=$PERSIST_IMAGE_NAME" >> compiler.env
+  variables:
+    BASE_IMAGE: $CSCS_REGISTRY_PATH/pika-spack-base:$SPACK_VERSION
+    DOCKERFILE: .gitlab/docker/Dockerfile.spack_compiler
+    DOCKER_BUILD_ARGS: '["BASE_IMAGE","ARCH","BOOST","COMPILER","CXXSTD","HWLOC"]'
+    CSCS_REBUILD_POLICY: always
+  artifacts:
+    reports:
+      dotenv: compiler.env
+
+###################################################################################################
+
+gcc11_debug_build:
+  stage: build
+  extends:
+    - .container-builder
+    - .variables_gcc11_config
+    - .cmake_variables_common
+  needs:
+    - gcc11_spack_image
+  before_script:
+    - build_type=`echo $BUILD_TYPE | tr '[:upper:]' '[:lower:]'`
+    - configuration=$CMAKE_COMMON_FLAGS-$CMAKE_FLAGS-$CI_COMMIT_SHORT_SHA
+    - configuration=${configuration//-D/}
+    - CONFIG_TAG=`echo $configuration | sha256sum - | head -c 16`
+    - export PERSIST_IMAGE_NAME=$CSCS_REGISTRY_PATH/pika-$compiler-$build_type-build:$CONFIG_TAG
+    - echo -e "PERSIST_IMAGE_NAME=$PERSIST_IMAGE_NAME" > build.env
+  variables:
+    DOCKERFILE: .gitlab/docker/Dockerfile.spack_build
+    CMAKE_FLAGS: "-DCMAKE_BUILD_TYPE=$BUILD_TYPE -DPIKA_WITH_CXX_STANDARD=$CXXSTD -DPIKA_WITH_MAX_CPU_COUNT=256 -DPIKA_WITH_MALLOC=system -DPIKA_WITH_SPINLOCK_DEADLOCK_DETECTION=ON"
+    DOCKER_BUILD_ARGS: '["BASE_IMAGE","CMAKE_COMMON_FLAGS","CMAKE_FLAGS"]'
+  artifacts:
+    reports:
+      dotenv: build.env
 
 ###################################################################################################
 
 .test_common:
   variables:
     SLURM_PARTITION: normal
+    SLURM_JOB_NUM_NODES: 1
+    SLURM_NTASKS: 1
 
 .test_common_mc:
   stage: test
   extends:
     - .test_common
     - .container-runner-daint-mc
-  variables:
-    SLURM_CONSTRAINT: mc
-    SLURM_PARTITION: normal
-    PROCESSOR_COUNT: 36
 
 .test_common_gpu:
   extends:
     - .test_common
     - .container-runner-daint-gpu
+
+gcc11_debug_test:
+  extends:
+    - .cmake_variables_common
+    - .variables_gcc11_config
+    - .test_common_mc
+  needs:
+    - gcc11_debug_build
+  script:
+    - spack build-env $spack_spec -- bash -c ctest --test-dir ${BUILD_DIR} --verbose \
+      -j$(nproc) --timeout 120 --output-on-failure --no-compress-output -R tests \
+      --exclude-regex "tests.unit.modules.executors.std_thread_scheduler"
```

Comment:
By the way, I think it would make sense already at this point to move the compile-only tests to the build/configure stage to build them on the rosa build farm instead of on daint.

Reply:
Yes please.